x-pack/plugins/apm/public/components/app/TransactionDetails/Transaction/WaterfallContainer/Waterfall/WaterfallItem.tsx

```diff
@@ -102,16 +96,61 @@ interface IWaterfallItemProps {
   onClick: () => any;
 }
 
-function Prefix({ item }: { item: IWaterfallItem }) {
-  if (item.docType !== 'transaction') {
+function PrefixIcon({ item }: { item: IWaterfallItem }) {
+  if (item.docType === 'span') {
+    // icon for database spans
+    const isDbType = item.span.span.type === 'db';
+    if (isDbType) {
+      return <EuiIcon type="database" />;
+    }
+
+    // omit icon for other spans
     return null;
   }
 
-  return (
-    <React.Fragment>
-      <EuiIcon type="merge" />{' '}
-    </React.Fragment>
-  );
+  // icon for RUM agent transactions
+  const isRumAgent = item.transaction.context.service.agent.name === 'js-base';
+  if (isRumAgent) {
+    return <EuiIcon type="globe" />;
+  }
+
+  // icon for other transactions
+  return <EuiIcon type="merge" />;
+}
+
+const StyledDuration = styled.span`
+  font-size: ${fontSizes.small};
+  color: ${colors.gray2};
+`;
```

Comment:
Would `EuiText` with a size and color prop work here? https://elastic.github.io/eui/#/display/text